controllers/rest/middleware_context.go

```diff
@@ -395,3 +403,94 @@ func convertProperties(properties map[string]interface{}, schema *openapi3.Schem
 	}
 	return properties, nil
 }
+
+//parseContextExtension takes the parameters from x-context andadds it to the context
+func parseContextExtension(c echo.Context, cc context.Context, params map[string]interface{}, entityFactory model.EntityFactory, api *RESTAPI) (context.Context, error) {
```

Comment:
this is duplicating functionality that is already in parse params. I feel like there must be a way to abstract parse params so it would work for both looking at parameters as well as parsing the values coming in from the extension